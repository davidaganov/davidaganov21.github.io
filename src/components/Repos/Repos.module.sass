.container
  width: 100%

.projects
  display: grid
  gap: 3rem
  width: 100%
  margin-top: 2rem
  @media (min-width: 769px)
    grid-template-columns: repeat(2, 1fr)

.tags
  display: flex
  flex-wrap: wrap
  gap: 1rem 1.5rem
  width: 100%
  margin-bottom: 2.5rem
  input
    &:focus-visible + .tag
      color: var(--color-accent)
    @media (min-width: 576px)
      &:focus-visible + .tagActive
        color: tomato
        border-color: tomato
  label
    min-height: 26px

.tag
  display: block
  padding: .7rem 1rem
  text-transform: uppercase
  border-style: solid
  border-width: .2rem
  color: var(--color-white-500)
  font: normal 1rem var(--second-font)
  transition: all .2s
  cursor: pointer
  &:hover
    border-color: var(--color-white)
    color: var(--color-white)
  &.tagActive
    border-color: var(--color-accent)
    color: var(--color-accent)
    @media (min-width: 576px)
      &:hover
        border-color: tomato
        color: tomato
