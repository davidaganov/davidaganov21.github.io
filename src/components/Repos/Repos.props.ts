import { DetailedHTMLProps, HTMLAttributes } from "react"
import { RepoProps } from "../../interfaces"

export interface ReposProps
  extends DetailedHTMLProps<HTMLAttributes<HTMLLIElement>, HTMLLIElement> {
  repos: RepoProps[]
}
