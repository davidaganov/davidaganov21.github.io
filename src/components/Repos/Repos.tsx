import { useState, useEffect } from "react"
import { useTranslation } from "react-i18next"
import { ReposProps } from "./Repos.props"
import styles from "./Repos.module.sass"
import cn from "classnames"

import { Card } from ".."
import { RepoProps } from "../../interfaces"

export const Repos = ({ repos }: ReposProps): JSX.Element => {
  const [selectTag, setSelectTag] = useState<string>("")
  const [sortRepos, setSortRepos] = useState<RepoProps[]>([])

  const { t } = useTranslation()

  useEffect(() => {
    setSortRepos(repos.filter((item) => item.topics.some((i) => [selectTag].includes(i))))
  }, [selectTag])

  const renderTags = () => {
    const topics: string[] = []

    repos.map((item) => {
      item.topics.map((tag) => {
        if (!topics.includes(tag)) topics.push(tag)
      })
    })

    const tags = topics.map((tag) => (
      <label
        htmlFor={`tag-${tag}`}
        key={tag}
      >
        <input
          className="visually-hidden"
          type="checkbox"
          name="tag"
          id={`tag-${tag}`}
          value={tag}
          checked={tag === selectTag}
          onChange={() => setSelectTag(tag === selectTag ? "" : tag)}
        />
        <span
          className={cn(styles.tag, {
            [styles.tagActive]: tag === selectTag
          })}
        >
          {tag}
        </span>
      </label>
    ))

    return (
      <fieldset className={styles.tags}>
        <legend className="visually-hidden">{t("projects.sort")}</legend>
        {tags}
      </fieldset>
    )
  }

  const renderItems = () => {
    const arr = sortRepos.length === 0 ? repos : sortRepos
    const items = arr.slice(0, 6).map((item) => {
      return (
        <Card
          key={item.id}
          card={item}
        />
      )
    })

    return <ul className={styles.projects}>{items}</ul>
  }

  const tags = renderTags()
  const projects = renderItems()

  return (
    <div className={styles.container}>
      <form id="selectTags">{tags}</form>
      {projects}
    </div>
  )
}
