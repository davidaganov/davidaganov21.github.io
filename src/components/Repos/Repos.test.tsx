import { render, screen } from "@testing-library/react"
import { Repos } from "./Repos"

describe("Repos component", () => {
  it("Repos renders", () => {
    render(<Repos />)

    expect(screen.getByText("#").closest("a")).toHaveAttribute("href", "#projects")
  })

  it("Repos snapshot", () => {
    const projects = render(<Repos />)

    expect(projects).toMatchSnapshot()
  })
})
