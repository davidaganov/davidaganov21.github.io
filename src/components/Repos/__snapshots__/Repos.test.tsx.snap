// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component Repos snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <form
          id="selectTags"
        >
          <fieldset
            class="tags"
          >
            <legend
              class="visually-hidden"
            >
              projects.sort
            </legend>
            <label
              for="tag-vue"
            >
              <input
                class="visually-hidden"
                id="tag-vue"
                name="tag"
                type="checkbox"
                value="vue"
              />
              <span
                class="tag"
              >
                vue
              </span>
            </label>
            <label
              for="tag-typescript"
            >
              <input
                class="visually-hidden"
                id="tag-typescript"
                name="tag"
                type="checkbox"
                value="typescript"
              />
              <span
                class="tag"
              >
                typescript
              </span>
            </label>
            <label
              for="tag-react"
            >
              <input
                class="visually-hidden"
                id="tag-react"
                name="tag"
                type="checkbox"
                value="react"
              />
              <span
                class="tag"
              >
                react
              </span>
            </label>
          </fieldset>
        </form>
        <ul
          class="projects"
        >
          <li
            class="project"
          >
            <h3
              class="title"
            >
              First repos
            </h3>
            <p
              class="description"
            >
              Lorem ipsum
            </p>
            <div
              class="bottom"
            >
              <a
                aria-label="projects.go_repo First repos"
                class="link inline-link"
                href="repo"
                rel="noreferrer"
                target="_blank"
              >
                projects.go_repo
              </a>
              <a
                aria-label="projects.go_demo First repos"
                class="link link--live inline-link"
                href="demo"
                rel="noreferrer"
                target="_blank"
              >
                projects.go_demo
              </a>
              <ul
                class="tags"
              >
                <li>
                  vue
                </li>
                <li>
                  typescript
                </li>
              </ul>
            </div>
          </li>
          <li
            class="project"
          >
            <h3
              class="title"
            >
              Second repos
            </h3>
            <p
              class="description"
            >
              Dolor sit
            </p>
            <div
              class="bottom"
            >
              <a
                aria-label="projects.go_repo Second repos"
                class="link inline-link"
                href="repo"
                rel="noreferrer"
                target="_blank"
              >
                projects.go_repo
              </a>
              <ul
                class="tags"
              >
                <li>
                  react
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <form
        id="selectTags"
      >
        <fieldset
          class="tags"
        >
          <legend
            class="visually-hidden"
          >
            projects.sort
          </legend>
          <label
            for="tag-vue"
          >
            <input
              class="visually-hidden"
              id="tag-vue"
              name="tag"
              type="checkbox"
              value="vue"
            />
            <span
              class="tag"
            >
              vue
            </span>
          </label>
          <label
            for="tag-typescript"
          >
            <input
              class="visually-hidden"
              id="tag-typescript"
              name="tag"
              type="checkbox"
              value="typescript"
            />
            <span
              class="tag"
            >
              typescript
            </span>
          </label>
          <label
            for="tag-react"
          >
            <input
              class="visually-hidden"
              id="tag-react"
              name="tag"
              type="checkbox"
              value="react"
            />
            <span
              class="tag"
            >
              react
            </span>
          </label>
        </fieldset>
      </form>
      <ul
        class="projects"
      >
        <li
          class="project"
        >
          <h3
            class="title"
          >
            First repos
          </h3>
          <p
            class="description"
          >
            Lorem ipsum
          </p>
          <div
            class="bottom"
          >
            <a
              aria-label="projects.go_repo First repos"
              class="link inline-link"
              href="repo"
              rel="noreferrer"
              target="_blank"
            >
              projects.go_repo
            </a>
            <a
              aria-label="projects.go_demo First repos"
              class="link link--live inline-link"
              href="demo"
              rel="noreferrer"
              target="_blank"
            >
              projects.go_demo
            </a>
            <ul
              class="tags"
            >
              <li>
                vue
              </li>
              <li>
                typescript
              </li>
            </ul>
          </div>
        </li>
        <li
          class="project"
        >
          <h3
            class="title"
          >
            Second repos
          </h3>
          <p
            class="description"
          >
            Dolor sit
          </p>
          <div
            class="bottom"
          >
            <a
              aria-label="projects.go_repo Second repos"
              class="link inline-link"
              href="repo"
              rel="noreferrer"
              target="_blank"
            >
              projects.go_repo
            </a>
            <ul
              class="tags"
            >
              <li>
                react
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
